src/components/PostDetails.tsx

```diff
@@ -1,107 +1,147 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
+import { Post } from '../types/Post';
+import { addComment, deleteComment, getComments } from '../utils/fetchClient';
+import { Comment, CommentData } from '../types/Comment';
+
+type Props = {
+  selectedPost: Post | null;
+  showForm: boolean;
+  setShowForm: (value: boolean) => void;
+};
+
+export const PostDetails: React.FC<Props> = ({
+  selectedPost,
+  showForm,
+  setShowForm = () => {},
```

Comment:
Setting a default value for `setShowForm` to an empty function can lead to unexpected behavior if the prop is not provided. Consider making this prop required without a default value.